### PR TITLE
added CI using gh-page and gtihub actions

### LINKED ISSUE #8: Add a GitHub action to deploy the demo to GitHub pages
Integrate this [action](https://github.com/marketplace/actions/deploy-to-github-pages) to automatically build the demo and deploy on the gh-pages branch.

### PR DESCRIPTION
**Pull request by ankiiitraj**
Fixes #8 
Github action triggers on every push! 
Check this here on the push to my fork: [ankiiitraj.github.io/react-copy-mailto](https://ankiiitraj.github.io/react-copy-mailto)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Build and Deploy
+on: [push]
```
Comment (prateek3255) [inline .github/workflows/main.yml:2]: Can you change this to -

```suggestion
on:
  push:
    branches:
      - master
```

Comment (ankiiitraj) [reply .github/workflows/main.yml:2]: resolving this!

**Comment by ankiiitraj** [comment]
Have a look now @prateek3255

**Comment by ankiiitraj** [comment]
Remainder to turn on the Github pages on this repo from settings. :)

**Comment by prateek3255** [comment]
> Remainder to turn on the Github pages on this repo from settings. :)

Yeah I know, Thanks for working on this @all-contributors please add ankiiitraj for tool

**Pull request merged**